src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/FrameResponseStream.cs

```diff
@@ -31,6 +34,7 @@ public override long Position
             set => throw new NotSupportedException();
         }
 
+        // REVIEW: Should Flush() throw?
```

Comment:
StreamWriter.Dispose calls Stream.Flush no matter what. You can prevent StreamWriter.Dispose from calling Stream.Write by either calling StreamWriter.FlushAsync immediately before or using "leaveOpen".